### PR TITLE
Number Input

### PR DESCRIPTION
**Pull request by antictor**
## Related issue
#208 

**Adds a new component `NumberInput` also known as 'number clicker'. I changed the name to NumberInput to be consistent with TextInput.**

Introduces a form element to allow users to enter numeric information either directly into the field or using arrow buttons to increase/decrease the number.

If the user tries to enter in text it is ignored. If the user tries to enter a number that is too big or too small it is ignored.

The developer can specify the increments the number should increase or decrease by. If the user does tries to enter a number between those gaps they are able to do so, so this should be enforced through form validation. Formik and Yup can capture this so when the user moves to another field it highlights the error.

number input

### CODE REVIEW

File: packages/css-framework/src/components/_numberinput.scss [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+.rn-numberinput__increase,
+.rn-numberinput__decrease {
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
+  width: 42px;
+  align-items: center;
+
```
Comment (m7kvqbe1) [inline packages/css-framework/src/components/_numberinput.scss:115]: Do we need this extra whitespace?

File: packages/docs-site/src/library/pages/components/forms/input.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Input
+title: Text Input
```
Comment (m7kvqbe1) [inline packages/docs-site/src/library/pages/components/forms/input.md:2]: Nit picking - should this be another PR? I guess it's a non-consequential change.

File: packages/reactapp/src/NumberForm.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
```
Comment (m7kvqbe1) [inline packages/reactapp/src/NumberForm.js:1]: I see you are using this app for testing? Because working with state in storybook is a real pain?

**Review by m7kvqbe1 — APPROVED** [review]
LGTM - just a couple of comments.

File: packages/css-framework/src/components/_numberinput.scss (hunk 1 above)
Comment (antictor) [reply packages/css-framework/src/components/_numberinput.scss:115]: don't need it it just helped me read the code and understand what was going on

File: packages/docs-site/src/library/pages/components/forms/input.md (hunk 2 above)
Comment (antictor) [reply packages/docs-site/src/library/pages/components/forms/input.md:2]: I changed it in this PR to differentiate it from the Number input. I though it was part of this pr. Good to see though that you are catching this :)

File: packages/reactapp/src/NumberForm.js (hunk 3 above)
Comment (antictor) [reply packages/reactapp/src/NumberForm.js:1]: I was getting an issue in Storybook in which it wasn't setting the field default value which seemed to be something to do with Storybook and not formik or my code so it was easiest to put an example in the test app.
